proxy/proxy.go

```diff
@@ -835,7 +835,15 @@ func (p *Proxy) lookupRoute(ctx *context) (rt *routing.Route, params map[string]
 // send a premature error response
 func (p *Proxy) sendError(c *context, id string, code int) {
 	addBranding(c.responseWriter.Header())
-	http.Error(c.responseWriter, http.StatusText(code), code)
+
+	text := http.StatusText(code)
+
+	c.responseWriter.Header().Set("Content-Length", fmt.Sprintf("%d", 1+len(text))) // count newline
+	c.responseWriter.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	c.responseWriter.Header().Set("X-Content-Type-Options", "nosniff")
+	c.responseWriter.WriteHeader(code)
+	fmt.Fprintln(c.responseWriter, text)
```

Comment:
https://github.com/golang/go/blob/b4c8b67adcd39da54f210bef5c201b1df8124d73/src/net/http/server.go#L2067-L2076

Reply:
alternatively: `c.responseWriter.Write([]byte(text))`, but it's also fine

Reply:
I have inlined `http.Error` implementation as it so that there is no implicit knowledge that it appends new line and length has to be adjusted

Reply:
I have changed it to write